config/routes.rb

```diff
@@ -0,0 +1,36 @@
+Rails.application.routes.draw do
+
+  # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
+  root to: 'homepages#index'
+
+# /product/1/order_items
+  resources :products do 
+    resources :order_items, only: [:index, :create, :show]
+    resources :reviews, only: [:create]
+  end
+
+  #Login
+  get "/auth/github", as: "github_login"
+  get "/auth/:provider/callback", to: "merchants#create"
```

Comment:
To of your tests fail because this isn't named.

```suggestion
  get "/auth/:provider/callback", to: "merchants#create", as: "auth_callback"
```